cmd/fleet/serve.go

```diff
@@ -557,6 +565,12 @@ the way that the Fleet server works.
 				pushProviderFactory := buford.NewPushProviderFactory()
 				mdmPushService = nanomdm_pushsvc.New(mdmStorage, mdmStorage, pushProviderFactory, nanoMDMLogger)
 				mdmCheckinAndCommandService = service.NewMDMAppleCheckinAndCommandService(ds)
+				appCfg.MDM.Enabled = true
+			}
+
+			// save the app config with the updated MDM.Enabled value
+			if err := ds.SaveAppConfig(context.Background(), appCfg); err != nil {
+				initFatal(err, "saving app config")
```

Comment:
@mna @gillespi314 before moving forward and figuring out a way to add tests for this, any thoughts/concerns about this approach?

Reply:
Yeah that makes sense to me, I can't think of an issue with saving the appconfig like that at startup. One thing though, it's already quite confusing that we have the `FleetConfig` and the `AppConfig`, now we will have `AppConfig.MDM.Enabled` and `FleetConfig.MDMApple.Enable`. I think we should have clearer names to differentiate their use, even if it means having a long annoying one (maybe `AppConfig.MDM.EnabledAndProperlyConfigured` or something like that? AIUI the `FleetConfig` one is currently required to enable all mdm endpoints (and is a flag that is meant to be removed going forward, I think), while `AppConfig` indicates if in addition to being enabled it is fully configured (at least for APNs/SCEP, I guess?).

Reply:
thank you very much! fully agree on naming, I will adjust.